### DISCUSSION TITLE
Form arguments before File arguments leads to 422 responses when uploading a file

### DISCUSSION THREAD

**Discussion by cfreksen** [fastapi/fastapi#12960] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

import uvicorn
from fastapi import FastAPI, File, Form

app = FastAPI()


@app.post("/foo")
async def foo(
    a: Annotated[int, Form()],
    b: Annotated[bytes, File()],
) -> dict[str, int]:
    return {"a": a, "len(b)": len(b)}


@app.post("/bar")
async def bar(
    b: Annotated[bytes, File()],
    a: Annotated[int, Form()],
) -> dict[str, int]:
    return {"a": a, "len(b)": len(b)}


if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=9099)
```

### Description

Calling the `/foo` endpoint with a file results in `422 Unprocessable Entity` because "Input should be a valid bytes", while doing the same for `/bar` works as intended. The expected behaviour is that `/foo` should behave as `/bar` does.

If I provide a string rather than a file as the `b` parameter, `/foo` accepts the request.

The following shell interaction demonstrates this with `curl`, `jq` is just used as a commandline json formatter.
```sh
$ curl -Ss -F "a=123" -F "b=@example.py" 'http://127.0.0.1:9099/bar' | jq
{
  "a": 123,
  "len(b)": 617
}
$ curl -Ss -F "a=123" -F "b=@example.py" 'http://127.0.0.1:9099/foo' | jq
{
  "detail": [
    {
      "type": "bytes_type",
      "loc": [
        "body",
        "b"
      ],
      "msg": "Input should be a valid bytes",
      "input": {
        "filename": "example.py",
        "file": {
          "_file": {},
          "_max_size": 1048576,
          "_rolled": false,
          "_TemporaryFileArgs": {
            "mode": "w+b",
            "buffering": -1,
            "suffix": null,
            "prefix": null,
            "encoding": null,
            "newline": null,
            "dir": null,
            "errors": null
          }
        },
        "size": 617,
        "headers": {
          "content-disposition": "form-data; name=\"b\"; filename=\"example.py\"",
          "content-type": "application/octet-stream"
        }
      }
    }
  ]
}
$ curl -Ss -F "a=123" -F "b=justSomeString" 'http://127.0.0.1:9099/foo' | jq
{
  "a": 123,
  "len(b)": 14
}
```

One can add `--trace-ascii %` to the curl command to see exactly what is send back and forth between `curl` and the FastAPI app, but I figured that would have been too verbose for this description.

### Operating System

Linux

### FastAPI Version

0.115.5

### Pydantic Version

2.9.2

### Python Version

Python 3.12.7

### Additional Context

Some other versions involved:
```
python-multipart version 0.0.9
curl version 8.11.0
```

**Comment by YuriiMotov** [comment, marked as answer]
This issue has been already addressed in this PR: https://github.com/fastapi/fastapi/pull/11194

But PR hasn't been merged yet..

**Reply by cfreksen** [reply]
Thank you for linking to that PR. I guess my Github searching skill wasn't as good as I thought, since I did not find that or discussion https://github.com/fastapi/fastapi/discussions/9116 when I did search.
